### DISCUSSION TITLE
How to generate my own Microsoft.Net.Compilers.Toolset package?

### DISCUSSION THREAD

**Discussion by sanmuru** [dotnet/roslyn#65271] [Q&A] [answered]
I add few features to CodeAnalysis.CSharp which pass all unit tests (including new unit tests for these new features), and follow the directions in [Building, Debugging and Testing on Windows](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Building%2C%20Debugging%2C%20and%20Testing%20on%20Windows.md) to deploy with command-line.
In the experimental instance, I test my new features and all things looks fine without build (no warnings or errors). But then, build fail and compiler complain about my unexpected operations.
Then I find this paragraph

> If you only install the VSIX, then the IDE will behave correctly (ie. new compiler and IDE behavior), but the Build operation or building from the command-line won't. To fix that, add a reference to the `Microsoft.Net.Compilers.Toolset` you built into your csproj. As shown below, you'll want to (1) add a nuget source pointing to your local build folder, (2) add the package reference, then (3) verify the Build Output of your project with a #error version included in your program.
![](https://user-images.githubusercontent.com/12466233/81205885-25252a80-8f80-11ea-9d75-268c7fe6f3ed.png)
![](https://user-images.githubusercontent.com/12466233/81205974-4128cc00-8f80-11ea-93ec-641d87662b12.png)
![](https://user-images.githubusercontent.com/12466233/81206129-7fbe8680-8f80-11ea-9438-acc0481a3585.png)

The question is, I quite sure there is no such directory like `packages\Release\Shipping` under `artifacts`, and of course I cannot find `Microsoft.Net.Compilers.Toolset` of version `v***-dev`.
Do I miss something?

**Comment by jasonmalinowski** [comment, marked as answer]
Hmm, that documentation might be out of date. You may have to do 'dotnet pack Roslyn.sln' to produce the packages.

I'm also a bit confused though, since for local development we also deploy a second VISX which should override the compiler used in projects. It's pretty fragile and we have a knack for breaking it, but it's odd the docs aren't mentioning that.

**Reply by sanmuru** [reply]
`dotnet pack` command do the job. Thanks!
